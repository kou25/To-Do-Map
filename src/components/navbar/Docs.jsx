import React, { Component } from 'react';
import { Document, Page, pdfjs } from 'react-pdf';
import docpdf from './Docs.pdf'
 

export default class Docs extends Component {
  constructor(props){
    super(props);
    pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

  this.state = {
    numPages: null,
    pageNumber: 1,
  }
}

  onDocumentLoad = ({ numPages }) => {
    this.setState({ numPages });
  }

  goToPrevPage = () =>{
  if(this.state.pageNumber>1){
    this.setState(state => ({ pageNumber: state.pageNumber - 1 }))}};
  goToNextPage = () =>{
  if(this.state.pageNumber<10){
    this.setState(state => ({ pageNumber: state.pageNumber + 1 }))}};



  render() {
    const { pageNumber, numPages } = this.state;

    return (
      <div className="container" style={{justifyContent:"center"}}>
         <nav style={{marginTop:"25px", marginLeft:"15%", display:"flex"}}>
          <button className="btn btn-danger shadow" onClick={this.goToPrevPage}>Prev</button>
          <button className="btn btn-success shadow ml-2 mr-5"onClick={this.goToNextPage}>Next</button>
          <span style={{marginLeft:"50%"}}><p style={{fontSize:"20px",fontWeight:"bold", paddingTop:"10px"}}>Page <span style={{color:"red"}}>{pageNumber}</span> of {numPages}</p></span>
        </nav>
        <div className="row" style={{ width: "800",paddingTop:"10px", justifyContent:"center" }}>
        <div className="card shadow p-5" style={{marginBottom:"100px"}} >
        <Document 
          file={docpdf}
          onLoadSuccess={this.onDocumentLoad}
        >
          <Page pageNumber={pageNumber} width={800}  />
        </Document>
        </div>
        </div>
      </div>
    );
  }
}