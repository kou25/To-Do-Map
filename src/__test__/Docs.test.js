import React from 'react';
import ReactDom from 'react-dom'
import Docs from '../components/navbar/Docs'
import { cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);
it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(<Docs/>,div);
})

it('matches snapshot', ()=>{
  const tree=renderer.create(
  <Docs/>).toJSON()
  expect(tree).toMatchSnapshot()
})