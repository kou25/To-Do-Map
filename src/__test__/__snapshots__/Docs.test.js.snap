// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <nav
    style={
      Object {
        "display": "flex",
        "marginLeft": "15%",
        "marginTop": "25px",
      }
    }
  >
    <button
      className="btn btn-danger shadow"
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      className="btn btn-success shadow ml-2 mr-5"
      onClick={[Function]}
    >
      Next
    </button>
    <span
      style={
        Object {
          "marginLeft": "50%",
        }
      }
    >
      <p
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
            "paddingTop": "10px",
          }
        }
      >
        Page 
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
          1
        </span>
         of 
      </p>
    </span>
  </nav>
  <div
    className="row"
    style={
      Object {
        "justifyContent": "center",
        "paddingTop": "10px",
        "width": "800",
      }
    }
  >
    <div
      className="card shadow p-5"
      style={
        Object {
          "marginBottom": "100px",
        }
      }
    >
      <div
        className="react-pdf__Document"
      >
        <div
          className="react-pdf__message react-pdf__message--loading"
        >
          Loading PDFâ€¦
        </div>
      </div>
    </div>
  </div>
</div>
`;
